app/scripts/modules/core/src/search/widgets/Search.tsx

```diff
@@ -0,0 +1,365 @@
+      if (text) {
+        if (this.hasModifier(value)) {
+          modifier = this.getModifier(value);
+        }
+
+        result = { modifier, text }
+      }
+    }
+
+    return result;
+  }
+
+  private getActiveFilterIndex(): number {
+    return this.filterTypes.findIndex((type: IFilterType) => type === this.state.activeFilter);
+  }
+
+  private getActiveFilterText(filterType: IFilterType): string {
+
+    let result: string;
+    const text = this.inputElement.value.trim();
+    if (this.hasModifier(text)) {
+      const modifier = this.getModifier(text);
+      const regex = new RegExp(modifier, 'i');
+      result = `${text.replace(regex, filterType.modifier)}`;
+    } else {
+      result = `${filterType.modifier}:${text}`;
+    }
+
+    return result;
+  }
+
+  private handleTagListUpdate(elements: HTMLElement[]): void {
+    this.tagElements = elements;
+  }
+
+  private refCallback(element: HTMLInputElement) {
+    this.inputElement = element;
+  }
+
+  private handleBlur(): void {
+    this.setFocusState(false);
+    this.setState({
+      isOpen: false
+    });
+  }
+
+  private handleChange(): void {
+    const value = this.inputElement.value.trim();
+    if (this.hasModifier(value)) {
+      const modifier = this.getModifier(value);
+      this.setState({
+        activeFilter: this.filterTypes.find((type: IFilterType) => type.modifier === modifier),
+        isOpen: true
+      });
+    } else if (value) {
+      this.setState({
+        isOpen: true
+      });
+    } else {
+      this.setState({
+        isOpen: false
+      });
+    }
+  }
+
+  private handleDelete(tag: ITag, focus: boolean): void {
+
+    this.setState({
+      isFocused: true,
+      tags: this.state.tags.filter((t: ITag) => tag !== t)
+    });
+
+    if (focus) {
+      this.inputElement.focus();
+    }
+  }
+
+  private handleFocus(): void {
+    this.setFocusState(true);
+  }
+
+  private navigateUp(): void {
+
+    if (this.state.isOpen) {
+      const { filterTypes } = this;
+      const activeIndex = this.getActiveFilterIndex();
+      if (activeIndex === 0) {
+        const activeFilter = filterTypes[filterTypes.length - 1];
+        this.inputElement.value = this.getActiveFilterText(activeFilter);
+        this.setState({
+          activeFilter,
+          isOpen: true
+        });
+      } else {
+        const activeFilter = filterTypes[activeIndex - 1];
+        this.inputElement.value = this.getActiveFilterText(activeFilter);
+        this.setState({
+          activeFilter,
+          isOpen: true
+        });
+      }
+    } else {
+      this.setState({
+        isOpen: true
+      })
+    }
+  }
+
+  private navigateDown(): void {
+
+    if (this.state.isOpen) {
+      const { filterTypes } = this;
+      const activeIndex = this.getActiveFilterIndex();
+      if (activeIndex === (filterTypes.length - 1)) {
+        const activeFilter = filterTypes[0];
+        this.inputElement.value = this.getActiveFilterText(activeFilter);
```

Comment:
feels like setting the value should be the responsibility of the `render` method, rather than something done within an event handler, but will defer to @jrsquared on that